### PR TITLE
Updated if block on wheel jobs to run on any label being added

### PR DESCRIPTION
**Pull request by rashidnhm**
**Context:**
The wheel building CI jobs can run automatically on a PR if an appropriate label is added (no need to push another commit). However, there is a bug where if you add a label to a job with no subsequent commits, and the label is not one of the `ci:` labels, then the job is skipped, which is an issue since it is a mandatory job.

**Description of the Change:**
This PR removes this if-statement, which means the job will run every time a label gets added to a PR, but that is fine since the job to determine if wheel CI builds should run is fairly light, and it avoids this bugs.

Furthermore, a small change to the error message [originally outlined here](https://github.com/PennyLaneAI/catalyst/pull/487#discussion_r1475099737) has been fixed.

**Benefits:**
No skipping of job on PRs.

**Possible Drawbacks:**
Runs more frequently and on every label being added to a PR (not that big of a drawback).

**Related GitHub Issues:**

### CODE REVIEW

**Comment by dime10** [comment]
Hi @rashidnhm, thanks for the fix. This doesn't mean that the wheels will run on every label change once the ci:build-wheels label is present, does it?

**Comment by rashidnhm** [comment]
> Hi @rashidnhm, thanks for the fix. This doesn't mean that the wheels will run on every label change once the ci:build-wheels label is present, does it?

Great question, and I believe it will. I'll need to look into how to change the logic since I can see that behavior not being desired

**Comment by dime10** [comment]
Would it make it easier if the wheel build was only triggered on `add label ci:build-wheels` events rather than on commits?

In practice it looks like once the label has been applied to a PR we end up building wheels on all commits even if there is still significant development left. When setting up the CI we tried to get the action runtime down as much as possible so that the developer can get fast feedback on a commit-level basis. The wheel builds, which are slower, are intended as a final check to ensure we didn't break the package distribution.

**Comment by rashidnhm** [comment]
I can definitely do that ... but that'd mean each time the check has to be run, the label has to be removed/re-added, wouldn't that add overhead for maintainers? Especially if there is a case where the author is unable to add/remove labels themselves.

Running the tests on each commit even after adding a label shouldn't be too much of an issue unless we are waiting a while to get the job picked up by a runner. Otherwise, as soon as a new commit is pushed, the older builds are immediately stopped.

**Comment by rashidnhm** [comment]
@dime10 I've made it so that if a label is added, the wheels are built only if the label is `ci:build-wheels`, for any other label, the job is not skipped but it will exit with success and not block merging of the PR (the wheel jobs themselves wouldn't run)

File: .github/workflows/check-for-wheel-build.yml
Comment (rauletorresc) [inline .github/workflows/check-for-wheel-build.yml:26]: lable -> label

**Review by rauletorresc — APPROVED** [review]
LGTM

**Review by maliasadi — APPROVED** [review]
:100:

**Review by dime10 — APPROVED** [review]
Thanks Rashid!

File: .github/workflows/check-for-wheel-build.yml
Comment (maliasadi) [reply .github/workflows/check-for-wheel-build.yml:26]:
```suggestion
      # then build wheels as long as the `ci:build-wheels` label is present
```

**Pull request merged**
